### PR TITLE
CT-2518 Capture Detainee/Ex-detainee/Probation service user as subjects

### PR DESCRIPTION
**Pull request by njpreddy**
Add 'Detainee', 'Ex-detainee' & 'Probation service user' as Offender SAR case subject types.

## Self-review checklist

* [x] (1) Quick stakeholder demo done OR
* [ ] (2) ...bug with before and after screenshots
* [x] (3) Tests passing
* [x] (4) Branch ready to be merged (not work in progress)
* [x] (5) No superfluous changes in diff
* [x] (6) No TODO's without new ticket numbers
* [x] (7) PR Prefixed with ticket number e.g. `CT-7654 ...`

### Screenshots
Subject type ordering on case creation page:

subject-type-order

Detainee case:

detainee

Ex-detainee caes:

ex-detainee

Probation service user case

probation-service-user

### Related JIRA tickets
https://dsdmoj.atlassian.net/secure/RapidBoard.jspa?rapidView=25&projectKey=CT&modal=detail&selectedIssue=CT-2518

### CODE REVIEW

File: app/decorators/case/sar/offender_decorator.rb [hunk 1]
```diff
@@ -5,7 +5,7 @@ def sar_response_address
   end
 
   def subject_type_display
-    object.subject_type.humanize.tr(' ', '-')
+    I18n.t('helpers.label.offender_sar.subject_type.'+object.subject_type)
```
Comment (Polsonby) [inline app/decorators/case/sar/offender_decorator.rb:8]: Tiny thing, sorry, please could we have spaces around the operator here?

```suggestion
    I18n.t('helpers.label.offender_sar.subject_type.' + object.subject_type)
```

File: app/models/case/sar/offender.rb [hunk 2]
```diff
@@ -47,8 +47,11 @@ def searchable_fields_and_ranks
...
 
   enum subject_type: {
-    offender: 'offender',
-    ex_offender: 'ex_offender',
+    offender: I18n.t('helpers.label.offender_sar.subject_type.offender'),
+    ex_offender: I18n.t('helpers.label.offender_sar.subject_type.ex_offender'),
+    detainee: I18n.t('helpers.label.offender_sar.subject_type.detainee'),
+    ex_detainee: I18n.t('helpers.label.offender_sar.subject_type.ex_detainee'),
+    probation_service_user: I18n.t('helpers.label.offender_sar.subject_type.probation_service_user'),
   }
```
Comment (Polsonby) [inline app/models/case/sar/offender.rb:55]: I think I see why you've done this, but I don't think the value is used in these enums - or am I missing something? :) 

If I'm not, for consistency, could we keep it the way it's done for other enums in the app? 

e.g. `ex_offender: 'ex_offender',` and so on?

File: app/decorators/case/sar/offender_decorator.rb (hunk 1 above)
Comment (njpreddy) [reply app/decorators/case/sar/offender_decorator.rb:8]: Yep, no problem

File: app/models/case/sar/offender.rb (hunk 2 above)
Comment (njpreddy) [reply app/models/case/sar/offender.rb:55]: Happy to change back to ex_offender: 'ex_offender', as the value is not used.

**Review by Polsonby — APPROVED** [review]
fab ;)

**Pull request merged**